src/gc/gc.cpp

```diff
@@ -36269,16 +36269,11 @@ CFinalize::ScanForFinalization (promote_func* pfn, int gen, BOOL mark_only_p,
 
                     assert (method_table(obj)->HasFinalizer());
 
-#ifndef FEATURE_REDHAWK
-                    if (method_table(obj) == pWeakReferenceMT || method_table(obj)->GetCanonicalMethodTable() == pWeakReferenceOfTCanonMT)
+                    if (GCToEEInterface::FinalizeIfWeakReference(obj))
```

Comment:
This is a general purpose callback that can be used to eagerly finalize any type during GC instead of waiting for it to be finalized later by the finalizer thread. It just happens that the current implementation is using it for finalizing weak references.

I would call it `bool EagerFinalization(Object* obj)` or something similar.